lib/cf_deployer/driver/dry_run.rb

```diff
@@ -4,6 +4,30 @@ class DryRun
 
       @@enabled = false
 
+      def self.enabled?
+        @@enabled
+      end
+
+      def self.run_with_value value, &block
+        previous_value = @@enabled
+        @@enabled = value
+        begin
+          block.call
+        rescue => e
+          raise e
+        ensure
+          @@enabled = previous_value
+        end
+      end
+
+      def self.enable_for &block
```

Comment:
Yes, 'enable_for' was added only for the sake of testing.  'DryRun.enabled?' is new as of this pull request - for the 'allow' that you describe to work, the guard method would need to be written to respect it as well.